File: CourseProjects/CourseProjects/VisualStudioProjects/ASP/AspWebFormsExamplesAndExercises/aspx_cs_samples/App_Code/CustomSettingsHandler.cs
using System;
using System.Configuration;
using System.Xml;

namespace AspxDemos {
	public class CustomSettingsHandler : IConfigurationSectionHandler {
		public object Create(object parent, object configContext, XmlNode section) {
			string tmpMaxThreads = "";
			string tmpDatabaseIP = "";
			foreach(XmlNode child in section.ChildNodes) {
				if(child is XmlElement) {
					XmlElement current = (XmlElement)child;
					if(current.Name.Equals("MaxThreads")) {
						tmpMaxThreads = current.InnerText;
					} else if(current.Name.Equals("DatabaseIP")) {
						tmpDatabaseIP = current.InnerText;
					}
				}
			}
			return new CustomSettingsVO(Convert.ToInt32(tmpMaxThreads.Trim()),tmpDatabaseIP);
		}
	}
}

File: CourseProjects/CourseProjects/VisualStudioProjects/ASP/AspWebFormsExamplesAndExercises/aspx_vb_samples/App_Code/VisualBasicWebApp/CustomSettingsHandler.vb
Imports Microsoft.VisualBasic

Imports System.Xml

Public Class CustomSettingsHandler
    Implements IConfigurationSectionHandler
    Public Function Create(ByVal parent As Object, ByVal configContext As Object, ByVal section As XmlNode) As Object Implements IConfigurationSectionHandler.Create
        Dim tmpMaxThreads As String = ""
        Dim tmpDatabaseIP As String = ""
        For Each child As XmlNode In section.ChildNodes
            If (TypeOf child Is XmlElement) Then
                Dim current As XmlElement = CType(child, XmlElement)
                If (current.Name.Equals("MaxThreads")) Then
                    tmpMaxThreads = current.InnerText
                ElseIf (current.Name.Equals("DatabaseIP")) Then
                    tmpDatabaseIP = current.InnerText
                End If
            End If
        Next
        Return New CustomSettingsVO(Convert.ToInt32(tmpMaxThreads.Trim()), tmpDatabaseIP)
    End Function
End Class
